media_driver/linux/common/cp/ddi/media_ddi_prot.cpp

```diff
@@ -328,9 +427,59 @@ VAStatus DdiMediaProtected::DetachProtectedSession(
 VAStatus DdiMediaProtected::ProtectedSessionExecute(
     VADriverContextP        ctx,
     VAProtectedSessionID    protected_session,
-    VABufferID              data)
+    VABufferID              bufId)
 {
-    return VA_STATUS_ERROR_UNIMPLEMENTED;
+    PDDI_MEDIA_CONTEXT             pMediaCtx            = nullptr;
+    PDDI_CP_CONTEXT                pCpCtx               = nullptr;
+    uint32_t                       ctxType              = 0;
+    PDDI_MEDIA_BUFFER              pBuf                 = nullptr;
+    VAStatus                       vaStatus             = VA_STATUS_SUCCESS;
+    void                           *pData               = nullptr;
+    int32_t                        ret                  = -1;
+    VAProtectedSessionExecuteBuffer *exec_buff          = nullptr;
+
+
+    DDI_CHK_NULL(ctx, "Null ctx.", VA_STATUS_ERROR_INVALID_CONTEXT);
+
+    pMediaCtx   = DdiMedia_GetMediaContext(ctx);
+    DDI_CHK_NULL(pMediaCtx, "Null pMediaCtx.", VA_STATUS_ERROR_INVALID_CONTEXT);
+
+    pCpCtx      = (PDDI_CP_CONTEXT)DdiMedia_GetContextFromContextID(ctx, protected_session, &ctxType);
+    DDI_CHK_NULL(pCpCtx, "Null pCpCtx.", VA_STATUS_ERROR_INVALID_CONTEXT);
+
+    pBuf = DdiMedia_GetBufferFromVABufferID(pMediaCtx, bufId);
+    DDI_CHK_NULL(pBuf, "Null pBuf.", VA_STATUS_ERROR_INVALID_BUFFER);
+
+    DdiMedia_MapBuffer(ctx, bufId, &pData);
+    DDI_CHK_NULL(pData, "Null pData.", VA_STATUS_ERROR_INVALID_BUFFER);
+
+    switch ((int32_t)pBuf->uiType)
+    {
+        // Protcted Buffer Types
+        case VAProtectedSessionExecuteBufferType:
+            exec_buff = (VAProtectedSessionExecuteBuffer *)pData;
+            ret = write(pCpCtx->fdMei, exec_buff->input.data, exec_buff->input.data_size);
+            if (ret == -1)
```

Comment:
should check more than -1 error code

Reply:
no handling for function id? how to know it is fw version cmd. need make logic clear to other reviewer.

Reply:
> should check more than -1 error code

Linux always returns -1 as error code, we have printed errno verbosely.